### PR TITLE
Milan/v3 agent config

### PR DESCRIPTION
**Pull request by milanaghera-tenable**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

**Test Configuration**:
* Python Version(s) Tested:
* Tenable.sc version (if necessary):

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: tenable/io/v3/vm/agent_config.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
+'''
+Agent Config
+============
+
+The following methods allow for interaction into the Tenable.io
+:devportal:`agent config <agent-config>` API endpoints.
+
+Methods available on ``tio.agent_config``:
```
Comment (dhvanilshah-tenable) [inline tenable/io/v3/vm/agent_config.py:8]: This needs to be updated in accordance with sphinx.

File: tenable/io/v3/vm/agent_config.py [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+.. rst-class:: hide-signature
+.. autoclass:: AgentConfigAPI
+    :members:
+'''
+from typing import Dict
+
+from ..base.endpoints.explore import ExploreBaseEndpoint
```
Comment (dhvanilshah-tenable) [inline tenable/io/v3/vm/agent_config.py:16]: Please replace it with absolute import.

File: tenable/io/v3/vm/agent_config.py [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+
+class AgentConfigAPI(ExploreBaseEndpoint):
+    '''
+    This will contain all methods related to agent config
+    '''
+    # _path: str = 'api/v3/agents'
+    _conv_json: bool = True
+
+    def edit(self, auto_unlink: int = None, software_update: bool = None,
```
Comment (dhvanilshah-tenable) [inline tenable/io/v3/vm/agent_config.py:26]: Please keep agent_id as the first parameter, and also remove the default value. Let's keep it required parameter.

File: tenable/io/v3/vm/agent_config.py [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+            payload['software_update'] = software_update
+        if auto_unlink and 0 < auto_unlink < 366:
+            payload['auto_unlink']['enabled'] = True
+            payload['auto_unlink']['expiration'] = auto_unlink
+        elif auto_unlink in [False, 0]:
+            payload['auto_unlink']['enabled'] = False
+        return self._put(f'scanners/{agent_id}/agents/config', json=payload)
+        # return self._put(f'{agent_id}/config', json=payload)
+
+    def details(self, agent_id: int = 1) -> Dict:
```
Comment (dhvanilshah-tenable) [inline tenable/io/v3/vm/agent_config.py:75]: Remove the default value from agent_config. Let's keep it required parameter.

File: tenable/io/v3/vm/agent_config.py [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+        '''
+        Returns the current agent configuration.
+
+        :devportal:`agent-config: details <agent-config-edit>`
+
+        Args:
+            agent_id (int, optional): The scanner ID.
```
Comment (dhvanilshah-tenable) [inline tenable/io/v3/vm/agent_config.py:82]: Doc needs to be updated.

File: tenable/io/v3/vm/agent_config.py [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+
+        Returns:
+            :obj:`Dict`:
+                Dictionary of the current settings.
+
+        Examples:
+            >>> details = tio.v3.vm.agent_config.details()
+            >>> pprint(details)
+        '''
+        return self._get(f'scanners/{agent_id}/agents/config')
```
Comment (dhvanilshah-tenable) [inline tenable/io/v3/vm/agent_config.py:92]: Please update the path as per it will be expected in v3.

File: tenable/io/v3/vm/agent_config.py [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+from typing import Dict
+
+from ..base.endpoints.explore import ExploreBaseEndpoint
+
+
+class AgentConfigAPI(ExploreBaseEndpoint):
+    '''
+    This will contain all methods related to agent config
+    '''
+    # _path: str = 'api/v3/agents'
```
Comment (dhvanilshah-tenable) [inline tenable/io/v3/vm/agent_config.py:23]: Please update the path as per it will be expected in v3.

File: tests/io/v3/vm/test_agent_config.py [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+    Test agent_config details endpoint
+    '''
+    test_response = {
+        'auto_unlink': {'enabled': True, 'expiration': 20},
+        'software_update': False
+    }
+    agent_id: int = 1
+    responses.add(
+        responses.GET,
+        re.compile(f'{BASE_URL}/scanners/{agent_id}/agents/config'),
```
Comment (dhvanilshah-tenable) [inline tests/io/v3/vm/test_agent_config.py:23]: Please update the path as per it will be expected in v3.

File: tests/io/v3/vm/test_agent_config.py [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+        'software_update': False
+    }
+    agent_id: int = 20
+    payload = {
+        'software_update': False,
+        'auto_unlink': {'enabled': True, 'expiration': 20}
+    }
+    responses.add(
+        responses.PUT,
+        re.compile(f'{BASE_URL}/scanners/{agent_id}/agents/config'),
```
Comment (dhvanilshah-tenable) [inline tests/io/v3/vm/test_agent_config.py:46]: Please update the path as per it will be expected in v3.

File: tenable/io/v3/vm/agent_config.py (hunk 6 above)
Comment (milanaghera-tenable) [reply tenable/io/v3/vm/agent_config.py:92]: Done

File: tenable/io/v3/vm/agent_config.py (hunk 7 above)
Comment (milanaghera-tenable) [reply tenable/io/v3/vm/agent_config.py:23]: Done.

File: tenable/io/v3/vm/agent_config.py (hunk 5 above)
Comment (milanaghera-tenable) [reply tenable/io/v3/vm/agent_config.py:82]: Done.

File: tenable/io/v3/vm/agent_config.py (hunk 4 above)
Comment (milanaghera-tenable) [reply tenable/io/v3/vm/agent_config.py:75]: Done.

File: tenable/io/v3/vm/agent_config.py (hunk 3 above)
Comment (milanaghera-tenable) [reply tenable/io/v3/vm/agent_config.py:26]: Done.

File: tenable/io/v3/vm/agent_config.py (hunk 2 above)
Comment (milanaghera-tenable) [reply tenable/io/v3/vm/agent_config.py:16]: Done

File: tenable/io/v3/vm/agent_config.py (hunk 1 above)
Comment (milanaghera-tenable) [reply tenable/io/v3/vm/agent_config.py:8]: Done.

File: tests/io/v3/vm/test_agent_config.py (hunk 9 above)
Comment (milanaghera-tenable) [reply tests/io/v3/vm/test_agent_config.py:46]: Done.

File: tests/io/v3/vm/test_agent_config.py (hunk 8 above)
Comment (milanaghera-tenable) [reply tests/io/v3/vm/test_agent_config.py:23]: Done.

File: tenable/io/v3/vm/agent_config.py (hunk 3 above)
Comment (dhvanilshah-tenable) [reply tenable/io/v3/vm/agent_config.py]: @milanaghera-tenable I think agent_id is still third parameter. Could you please move it as the first parameter?

File: tenable/io/v3/vm/agent_config.py (hunk 5 above)
Comment (dhvanilshah-tenable) [reply tenable/io/v3/vm/agent_config.py]: This is still not updated.

**Review by dhvanilshah-tenable — APPROVED** [review]

File: tenable/io/v3/vm/agent_config.py (hunk 5 above)
Comment (milanaghera-tenable) [reply tenable/io/v3/vm/agent_config.py]: Done.

File: tenable/io/v3/vm/agent_config.py (hunk 3 above)
Comment (milanaghera-tenable) [reply tenable/io/v3/vm/agent_config.py]: resolved.

File: tenable/io/v3/vm/plugins.py [hunk 10]
```diff
@@ -5,7 +5,7 @@
 The following methods allow for interaction into the Tenable.io
 :devportal:`plugins <plugins>` API endpoints.
 
-Methods available on ``tio.plugins``:
+Methods available on ``tio.v3.vm.plugins``:
```
Comment (tomeara) [inline tenable/io/v3/vm/plugins.py:8]: Why is this included with this PR? I would keep this PR focused only on Agent Config

**Review by tomeara — APPROVED** [review]

File: tenable/io/v3/vm/plugins.py (hunk 10 above)
Comment (milanaghera-tenable) [reply tenable/io/v3/vm/plugins.py:8]: Ted. Sure, will ensure this. PR related to plugins is merged, so I thought to include this change in this PR itself as this was a small change.
